packages/text-input/src/index.tsx

```diff
@@ -1,33 +1,82 @@
 import * as React from 'react'
-import {TextInputAppearanceType, TextInputProps} from './types'
+import {TextInputAppearanceType, TextInputProps, AffixProps} from './types'
 import * as S from './styled'
 import {
   getTextSizeForControlHeight,
   getBorderRadiusForControlHeight,
   useDefaults
 } from '@smashing/theme'
 
+const Affix: React.FC<AffixProps> = ({
+  affix,
+  component: IconComponent,
+  disabled,
+  height,
+  inputRef,
+  invalid
+}) => (
+  <React.Fragment>
+    {affix === 'before' ? (
+      <S.InputBefore invalid={invalid} disabled={disabled} height={height}>
+        <IconComponent inputRef={inputRef} />
+      </S.InputBefore>
+    ) : (
+      <S.InputAfter
+        invalid={invalid}
+        disabled={disabled}
+        height={height}
+        string={typeof IconComponent === 'string'}
+      >
+        {typeof IconComponent === 'string' ? (
+          IconComponent
+        ) : (
+          <IconComponent inputRef={inputRef} />
+        )}
+      </S.InputAfter>
+    )}
+  </React.Fragment>
+)
+
 const TextInput: React.FC<TextInputProps> = ({
   children,
   innerRef,
+  before,
```

Comment:
consider renaming to `prefix`/`postfix` or `affixBefore`/`affixAfter`, it's too general name